sdk/search/search/review/search.api.md

```diff
@@ -10,6 +10,48 @@ import { PipelineOptions } from '@azure/core-http';
 import { ServiceClientCredentials } from '@azure/core-http';
 import { WebResource } from '@azure/core-http';
 
+// @public
+export interface AccessCondition {
+    ifMatch?: string;
+    ifNoneMatch?: string;
+}
+
+// @public
+export interface Analyzer {
+    name: string;
+    odatatype: "Analyzer";
+}
+
+// @public
+export interface AnalyzeRequest {
+    analyzer?: AnalyzerName;
+    charFilters?: CharFilterName[];
+    text: string;
+    tokenFilters?: TokenFilterName[];
+    tokenizer?: TokenizerName;
+}
+
+// @public
+export interface AnalyzeResult {
+    tokens: TokenInfo[];
+}
+
+// @public
+export type AnalyzerName = 'ar.microsoft' | 'ar.lucene' | 'hy.lucene' | 'bn.microsoft' | 'eu.lucene' | 'bg.microsoft' | 'bg.lucene' | 'ca.microsoft' | 'ca.lucene' | 'zh-Hans.microsoft' | 'zh-Hans.lucene' | 'zh-Hant.microsoft' | 'zh-Hant.lucene' | 'hr.microsoft' | 'cs.microsoft' | 'cs.lucene' | 'da.microsoft' | 'da.lucene' | 'nl.microsoft' | 'nl.lucene' | 'en.microsoft' | 'en.lucene' | 'et.microsoft' | 'fi.microsoft' | 'fi.lucene' | 'fr.microsoft' | 'fr.lucene' | 'gl.lucene' | 'de.microsoft' | 'de.lucene' | 'el.microsoft' | 'el.lucene' | 'gu.microsoft' | 'he.microsoft' | 'hi.microsoft' | 'hi.lucene' | 'hu.microsoft' | 'hu.lucene' | 'is.microsoft' | 'id.microsoft' | 'id.lucene' | 'ga.lucene' | 'it.microsoft' | 'it.lucene' | 'ja.microsoft' | 'ja.lucene' | 'kn.microsoft' | 'ko.microsoft' | 'ko.lucene' | 'lv.microsoft' | 'lv.lucene' | 'lt.microsoft' | 'ml.microsoft' | 'ms.microsoft' | 'mr.microsoft' | 'nb.microsoft' | 'no.lucene' | 'fa.lucene' | 'pl.microsoft' | 'pl.lucene' | 'pt-BR.microsoft' | 'pt-BR.lucene' | 'pt-PT.microsoft' | 'pt-PT.lucene' | 'pa.microsoft' | 'ro.microsoft' | 'ro.lucene' | 'ru.microsoft' | 'ru.lucene' | 'sr-cyrillic.microsoft' | 'sr-latin.microsoft' | 'sk.microsoft' | 'sl.microsoft' | 'es.microsoft' | 'es.lucene' | 'sv.microsoft' | 'sv.lucene' | 'ta.microsoft' | 'te.microsoft' | 'th.microsoft' | 'th.lucene' | 'tr.microsoft' | 'tr.lucene' | 'uk.microsoft' | 'ur.microsoft' | 'vi.microsoft' | 'standard.lucene' | 'standardasciifolding.lucene' | 'keyword' | 'pattern' | 'simple' | 'stop' | 'whitespace';
```

Comment:
Is this extensible? Users can define their own analyzers via `CustomAnalyzer`, which introduces new names into this type at run-time.

Reply:
in theory you could add `| string`, but I don't believe TS does the right thing here. I think the way we were thinking of modeling cases like this is we'd have the API itself just take `string` but then provide a string enum of known values they can use.

Reply:
That works too. In Track 1 .NET, the main reason we introduced extensible enums was for discoverability of the different options via Intellisense, not safety (since any string can effectively be cast to that type).

Reply:
I tweaked this so we model the names of stuff as "string" but provide doc references to a well known enum. @bterlson figured out a clever hack to try to get the TS language service to show the suggested known values, but I'm a little wary to go there, since it won't help JS customers anyway.